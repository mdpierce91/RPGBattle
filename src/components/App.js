import React, { useReducer } from 'react';
import AppContext from '../context/app-context';
import ConfigureStore from '../store/configureStore';
import AppRouter from '../routers/AppRouter';
import authReducer from '../reducers/auth';
import unitsReducer from '../reducers/units';
import teamReducer from '../reducers/team';

export const App = () => {
    const [auth, authDispatch] = useReducer(authReducer, {});
	const [units, unitsDispatch] = useReducer(unitsReducer, []);
    const [team, teamDispatch] = useReducer(teamReducer, []);
    const store = ConfigureStore();
    return (
        <AppContext.Provider store={store} value={{units, team, auth, unitsDispatch, teamDispatch, authDispatch}}>
            <AppRouter/>
        </AppContext.Provider>
    )
}
export { App as default };