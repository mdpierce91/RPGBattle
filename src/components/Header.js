import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/">
                    <h1>RPG Battle</h1>
                </Link>
                <div>
                    <button className="button button--link">Settings</button>
                    <button className="button button--link button--logout" onClick={startLogout}>Logout</button>
                </div>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);