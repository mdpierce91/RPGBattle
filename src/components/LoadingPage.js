import React from 'react';

export const LoadingPage = () => (
    <div className="loader">
     <img className="loader__image" src="/images/loader.gif" alt=""/>
    </div>
);

export default LoadingPage;