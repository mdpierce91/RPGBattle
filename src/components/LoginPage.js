import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

/**
 * This presents the login buttons to the user.
 * Currently only allows login with Google.
 * @param {object} props - Props for the constructor
 */
export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">RPGBattle</h1>
            <p>It's time to Battle</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);