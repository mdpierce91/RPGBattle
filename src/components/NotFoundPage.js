import React from 'react';
import { NavLink } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        404 - <NavLink exact={true} to="/">Go Home</NavLink>
    </div>
);
export default NotFoundPage;