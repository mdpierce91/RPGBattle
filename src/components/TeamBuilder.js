import React from 'react';
import UnitList from './UnitList';

const TeamBuilder = () => {
    return (
        <div>
            
            <UnitList />
            
        </div>
    )
}

export { TeamBuilder as default };