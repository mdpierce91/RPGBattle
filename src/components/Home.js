import React from 'react';
import LoginPage from './LoginPage';

const Home = () => {
    return (
        <div className="container home-container">
            <LoginPage/>
        </div>
    );
}

export { Home as default };