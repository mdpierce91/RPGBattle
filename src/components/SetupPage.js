import React from 'react';
import TeamBuilder from './TeamBuilder';

const SetupPage = () => {
    return (
        <div>
            <TeamBuilder />
            <button>Find Battle!</button>
        </div>
    );
}

export { SetupPage as default };