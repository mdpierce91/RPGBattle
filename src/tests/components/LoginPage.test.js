import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

// Enzyme.configure({ adapter: new Adapter() });
let startLogin, wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin} />);
}) 

test('should render the login page correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
    wrapper.find('.button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});
