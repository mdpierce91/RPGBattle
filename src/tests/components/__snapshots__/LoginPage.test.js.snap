// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the login page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      RPGBattle
    </h1>
    <p>
      It's time to Battle
    </p>
    <button
      className="button"
      onClick={[MockFunction]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
